encoder/vaapiencoder_base.cpp

```diff
@@ -454,8 +454,8 @@ void VaapiEncoderBase::fill(VAEncMiscParameterRateControl* rateControl, uint32_t
 #if VA_CHECK_VERSION(0, 39, 4)
     rateControl->rc_flags.bits.temporal_id = temproalID;
 #endif
-    rateControl->bits_per_second = ((temproalID == 0) ?
-        m_videoParamCommon.rcParams.bitRate : m_videoParamCommon.temporalLayers.bitRate[temproalID-1]);
+    rateControl->bits_per_second =
+        (temproalID + 1) == m_videoParamCommon.temporalLayers.numLayers ? bitRate() : m_videoParamCommon.temporalLayers.bitRate[temproalID];
```

Comment:
better change it to temproalID == (m_videoParamCommon.temporalLayers.numLayers - 1) and add some comments